### DISCUSSION TITLE
Why are there no abnormalities in Primitives Example and there is no display on the development board screen

### DISCUSSION THREAD

**Discussion by zhangqiang199226** [nanoframework/Home#1271] [Q&A]
![image](https://user-images.githubusercontent.com/74021799/231697224-303af1cc-823d-40e0-ac91-a68d98443c48.png)
![3803cfe81d0ec9f2a4b4f8d80d94a90](https://user-images.githubusercontent.com/74021799/231697482-041e26b0-1de4-4f04-9813-01e4fad13509.jpg)

**Comment by Ellerbach** [comment]
Console.WriteLine won't write on the screen. Use the primitives and follow the examples.

**Reply by zhangqiang199226** [reply]
#define ESP32 // Comment this in if for the target platform!
// !!!-----------------------------------------------------------------------------!!!
using System;
using System.Threading;
using nanoFramework.Hardware.Esp32;
using nanoFramework.UI;
using Primitives.SimplePrimitives;

namespace Primitives
{
    public class Program
    {
        public static void Main()
        {
            int delayBetween = 1100;

//#if ESP32   // This is an example mapping, work them out for your needs!
            int backLightPin = 5;
            int chipSelect = 22;
            int dataCommand = 21;
            int reset = 18;
            // Add the nanoFramework.Hardware.Esp32 to the solution
            Configuration.SetPinFunction(25, DeviceFunction.SPI1_MISO);
            Configuration.SetPinFunction(23, DeviceFunction.SPI1_MOSI);
            Configuration.SetPinFunction(19, DeviceFunction.SPI1_CLOCK);

            // Adjust as well the size of your screen and the position of the screen on the driver
            uint mun =  DisplayControl.Initialize(new SpiConfiguration(1, chipSelect, dataCommand, reset, backLightPin), 
                new ScreenConfiguration(0, 0, 160, 120), 262144u);
            
            // Depending on you ESP32, you may also have to use either PWM either GPIO to set the backlight pin mode on


            //new GpioController().OpenPin(backLightPin, PinMode.Output);
            //new GpioController().Write(backLightPin, PinValue.High);

//#elif STM32F769I_DISCO // This is an example (working) button map, work the actual pins out for your need!
//            //WARNING: Invalid pin mappings will never be returned, and may need you to reflash the device!
//            DisplayControl.Initialize(new SpiConfiguration(), new ScreenConfiguration());
//#else
//            throw new System.Exception("Unknown display mapping!");
//#endif

            // Get full screen bitmap from displayControl to draw on.yyyyyyyyyyyyyyyyyyy
            Bitmap fullScreenBitmap = DisplayControl.FullScreen;  

            fullScreenBitmap.Clear();

            Font DisplayFont = Resource.GetFont(Resource.FontResources.SegoeUIRegular12);

            while (true)
            {
                RandomDrawLine rdlt = new RandomDrawLine(fullScreenBitmap, DisplayFont);
                Thread.Sleep(delayBetween);

                RotateImage ri = new RotateImage(fullScreenBitmap, DisplayFont);
                Thread.Sleep(delayBetween);

                ColourGradient colourGradient = new ColourGradient(fullScreenBitmap, DisplayFont);
                Thread.Sleep(delayBetween);

                Colours ColourExample = new Colours(fullScreenBitmap, DisplayFont);
                Thread.Sleep(delayBetween);

                PagedText pt = new PagedText(fullScreenBitmap, DisplayFont);
                Thread.Sleep(delayBetween);

                BouncingBalls bb = new BouncingBalls(fullScreenBitmap, DisplayFont);
                Thread.Sleep(delayBetween);

                TileImage ti = new TileImage(fullScreenBitmap, DisplayFont);
                Thread.Sleep(delayBetween);

                StretchImage si = new StretchImage(fullScreenBitmap, DisplayFont);
                Thread.Sleep(delayBetween);

                SetPixels sp = new SetPixels(fullScreenBitmap, DisplayFont);
                Thread.Sleep(delayBetween);

                FontExamples fe = new FontExamples(fullScreenBitmap);
                Thread.Sleep(delayBetween);

                RandomRectangles rr = new RandomRectangles(fullScreenBitmap, DisplayFont);
                Thread.Sleep(delayBetween);

                SliceScaling9 ss = new SliceScaling9(fullScreenBitmap, DisplayFont);
                Thread.Sleep(delayBetween);

                RandomClipping rc = new RandomClipping(fullScreenBitmap, DisplayFont);
                Thread.Sleep(delayBetween);

                MatrixRain mr = new MatrixRain(fullScreenBitmap);
               // Thread.Sleep(Timeout.Infinite);
            }
        }
    }
}

**Reply by zhangqiang199226** [reply]
There is no abnormality with using this, but the screen only has backlight and no graphics.

**Reply by Ellerbach** [reply]
With which image have you flashed your ESP32?

**Reply by zhangqiang199226** [reply]
ESP_WROVER_KIT

**Reply by zhangqiang199226** [reply]
1.8.1.110

**Reply by Ellerbach** [reply]
Son you already have a driver implemented in there. You should not use the generic driver but rather leave it to null
